src/hough.rs

```diff
@@ -42,7 +43,9 @@ pub fn detect_lines(image: &GrayImage, options: LineDetectionOptions) -> Vec<Pol
     let rmax = ((width * width + height * height) as f64).sqrt() as i32;
 
     // Measure angles in degrees, and use bins of width 1 pixel and height 1 degree.
-    let mut acc: ImageBuffer<Luma<u32>, Vec<u32>> = ImageBuffer::new(rmax as u32, 180u32);
+    // We use the convention that distances are positive for angles in (0, 180] and
+    // negative for angles in [180, 360).
+    let mut acc: ImageBuffer<Luma<u32>, Vec<u32>> = ImageBuffer::new(2 * rmax as u32 + 1, 180u32);
```

Comment:
This introduces an index error in the unsafe below, I think. If `rmax = 1.6` then:

`2 + rmax as u32 + 1 = 3`

but at the same time, we can get

`r=1.4 => d as u32 = (r + rmax) as u32 = (1.4 + rmax) as u32 = 3`
`d = 3 <= 2*rmax = 3.2`

Edit: nvm, rmax is `i32`, I forgot cast happens before.

Reply:
Sorry.